### DISCUSSION TITLE
Trouble building documentation (missing shared libraries on AMI, sphinx)

### DISCUSSION THREAD

**Discussion by willsmithorg** [autogluon/autogluon#1451] [Q&A] [answered]
I'm trying to build the documentation, using the instructions in docs/README.md

I'm using a default Amazon Linux AMI (Amazon Linux 2 AMI (HVM) - Kernel 5.10).

First I got:
```
OSError: libquadmath.so.0: cannot open shared object file: No such file or directory
```

Which I traced back to no gcc on the AMI, so I ran
```
sudo yum install gcc
```

Then next time I tried to rebuild the documentation, it got further but soon errored with:
```
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

which I also solved with 
```
sudo yum install mesa-libGL
```

You might want to mention these 2 dependencies are required to build the docs.

**Comment by willsmithorg** [comment]
Now I'm getting a warning that I can't run some notebooks without a GPU.
Is there some way to only build part of the documentation?  In my case I just want to build the documentation at api/autogluon.features.html, since I've changed the docstrings in the source.

**Comment by willsmithorg** [comment]
As a workaround I renamed most of the .md files in doc/tutorials to .md.hidden

Now, when I try to execute 'bash build_doc.sh' I get

```
Running Sphinx v4.3.2
WARNING: while setting up extension jupyter_sphinx.execute: `jupyter-sphinx` was initialized with the `jupyter_sphinx.execute` sub-module. Replace this with `jupyter_sphinx`. Initializing with `jupyter_sphinx.execute` will be removed in version 0.3
```

This is a shame, I was hoping to contribute to documentation, but now I'm stuck.  I see you have a CI process to automatically build the documentation, but I don't want to have to invoke/wait for that to test whether my markdown renders correctly as HTML.

**Reply by Innixma** [reply]
Thanks for noting these issues. This is very helpful for us to make documentation contributions easier (both for internal and external).

I was able to reproduce your error after deleting the tutorials/* files.

Full error:

```
Running Sphinx v4.3.2
WARNING: while setting up extension jupyter_sphinx.execute: `jupyter-sphinx` was initialized with the `jupyter_sphinx.execute` sub-module. Replace this with `jupyter_sphinx`. Initializing with `jupyter_sphinx.execute` will be removed in version 0.3
making output directory... done

Exception occurred:
  File "/home/ubuntu/remote2/autogluon/docs/_build/rst/conf.py", line 140, in setup
    app.add_javascript('d2l.js')
AttributeError: 'Sphinx' object has no attribute 'add_javascript'
The full traceback has been saved in /tmp/sphinx-err-i31dskn2.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

I've done a deep dive and fixed the various issues surrounding this in #1452. Hopefully the [updated docs/README.md](https://github.com/awslabs/autogluon/blob/improve_doc_setup/docs/README.md) will work for you (although the README will only work on the PR branch until it is merged, as I did various other optimizations / simplifications of the CI / Doc Build logic).

**Reply by Innixma** [reply]
@willsmithorg #1452 has been merged, let me know if you get a chance to try out building the docs with the new README, hopefully it should work fine.

**Comment by Innixma** [comment, marked as answer]
#1452 has been merged, providing a new README to building the docs which avoids the problems mentioned in this discussion.
